RebornCore/src/main/java/reborncore/common/BaseBlockEntityProvider.java

```diff
@@ -49,27 +50,29 @@ public Optional<ItemStack> getDropWithContents(World world, BlockPos pos, ItemSt
 		if (blockEntity == null) {
 			return Optional.empty();
 		}
+
 		ItemStack newStack = stack.copy();
-		NbtCompound blockEntityData = blockEntity.createNbt();
-		stripLocationData(blockEntityData);
-		if (!newStack.hasNbt()) {
-			newStack.setNbt(new NbtCompound());
-		}
-		newStack.getOrCreateNbt().put("blockEntity_data", blockEntityData);
+		newStack.applyComponentsFrom(blockEntity.createComponentMap());
+//		NbtCompound blockEntityData = blockEntity.createNbt(world.getRegistryManager());
+//		stripLocationData(blockEntityData);
+//		if (newStack.getComponents() == ComponentMap.EMPTY) {
+//			newStack.setNbt(new NbtCompound());
+//		}
+//		newStack.getOrCreateNbt().put("blockEntity_data", blockEntityData);
 		return Optional.of(newStack);
 	}
 
-	@Override
-	public void onPlaced(World worldIn, BlockPos pos, BlockState state, LivingEntity placer, ItemStack stack) {
-		if (stack.hasNbt() && stack.getOrCreateNbt().contains("blockEntity_data")) {
-			BlockEntity blockEntity = worldIn.getBlockEntity(pos);
-			if (blockEntity == null) { return; }
-			NbtCompound nbt = stack.getOrCreateNbt().getCompound("blockEntity_data");
-			injectLocationData(nbt, pos);
-			blockEntity.readNbt(nbt);
-			blockEntity.markDirty();
-		}
-	}
+//	@Override
```

Comment:
For commented out stuff can you put a `//TODO 1.20.5` comment on it? Should make it a bit easier to find/deal with later.

Im quite happy to comment stuff out to make it somewhat work though 👍

Reply:
Yeah, TODO is a nice idea. This one left commented intentionally. To check if itemBlock will properly restore inventory component